runtime/drivers/snowflake/snowflake.go

```diff
@@ -80,8 +82,18 @@ func (d driver) Open(instanceID string, config map[string]any, st *storage.Clien
 		return nil, err
 	}
 
-	// actual db connection is opened during query
+	if conf.DSN == "" {
+		return nil, fmt.Errorf("dsn not provided")
+	}
+
+	// Open DB handle
+	db, err := sqlx.Open("snowflake", conf.DSN)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
+
 	return &connection{
+		db:               db,
```

Comment:
Keeping the connection permanently open may not be desirable since it's used so rarely. Or at least also use this connection in `warehouse.go`